src/core/Akka.Persistence.TestKit/JournalInterceptors.cs

```diff
@@ -0,0 +1,99 @@
+﻿namespace Akka.Persistence.TestKit
+{
+    using System;
+    using System.Threading.Tasks;
+
+    internal static class JournalInterceptors
+    {
+        internal class Noop : IJournalInterceptor
+        {
+            public static readonly IJournalInterceptor Instance = new Noop();
+
+            public Task InterceptAsync(IPersistentRepresentation message) => Task.FromResult(true);
+        }
+
+        internal class Failure : IJournalInterceptor
+        {
+            public static readonly IJournalInterceptor Instance = new Failure();
+
+            public Task InterceptAsync(IPersistentRepresentation message) => throw new TestJournalFailureException(); 
+        }
+
+        internal class Rejection : IJournalInterceptor
+        {
+            public static readonly IJournalInterceptor Instance = new Rejection();
+
+            public Task InterceptAsync(IPersistentRepresentation message) => throw new TestJournalRejectionException(); 
+        }
+        
+        internal class Delay : IJournalInterceptor
```

Comment:
Is the goal of this `IJournalInterceptor` to simulate latency?

Reply:
Yes, like if there is some retry strategy which will make several attempts before fail.

It was eady to add...